### PR TITLE
ADD : Responsiveness to main page #28

### PR DESCRIPTION
**Pull request by RaiDevX8**
Here's a revised description for your pull request (PR) based on the changes you've made:

---

## 📋 Description
#28 
This PR focuses on improving the responsiveness of the existing components in the application, ensuring a better user experience across various screen sizes.

## 🔨 Changes Made

- Implemented responsive design adjustments for the **SideDrawer** component to enhance usability on mobile devices.
- Made modifications to the layout and styling of other UI elements to ensure they adapt appropriately to different screen resolutions.

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings.
- [x] I have tested it locally and it works fine.
- [ ] Any dependent changes have been merged and published in downstream modules.

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [x] Bug fix (non-breaking change which fixes an issue) 🐛
- [x] New feature (non-breaking change which adds functionality) ✨
- [x] UI enhancement (non-breaking change which enhances UI) 🎨
- [ ] Documentation update 📚

## 🤝 Related Issues

- Fixes #

## 📸 Screenshots (if applicable)

Screenshot 2024-10-06 221235
Screenshot 2024-10-06 221245
Screenshot 2024-10-06 221307

---

---

Feel free to adjust any part of the description as needed!

### CODE REVIEW

**Comment by RaiDevX8** [comment]
It's level 2 labelled assignment ig

File: frontend/src/components/SingleChat.js [hunk 1]
```diff
@@ -221,35 +217,34 @@ const SingleChat = ({ fetchAgain, setFetchAgain }) => {
             )}
 
             <FormControl
-              onKeyDown={sendMessage}
+              onKeyDown={(event) => event.key === "Enter" && sendMessage()}
```
Comment (kartikayasijaa) [inline frontend/src/components/SingleChat.js:220]: verify the key down event in sendMessage function

**Comment by kartikayasijaa** [comment]
@Sibam-Paul Can you test this?

File: frontend/src/components/SingleChat.js (hunk 1 above)
Comment (RaiDevX8) [reply frontend/src/components/SingleChat.js:220]: The original code did not have a send button. By pressing Enter, the message was sent, but it didn't work on mobile screens, so I added a new send button. @kartikayasijaa

**Comment by Sibam-Paul** [comment]
@kartikayasijaa yup ok

File: frontend/src/components/SingleChat.js (hunk 1 above)
Comment (Sibam-Paul) [reply frontend/src/components/SingleChat.js:220]: Means in large screen ??? no

Comment (RaiDevX8) [reply frontend/src/components/SingleChat.js:220]: Both have send button

**Comment by Sibam-Paul** [comment]
@RaiDevX8 nice job

**Pull request merged**